### PR TITLE
Add button edit task to the task.component

### PR DESCRIPTION
**Pull request by MoubarkiS**
Add task modification functionality

- Added an "Edit" button in the task component to open an edit modal.
- Created the edit modal for tasks, including a form to modify the task name, date, and importance level.
- Updated TaskService to support task updates.
- Added CSS styles for the task edit modal.

### CODE REVIEW

**Comment by MoubarkiS** [comment]
Add task editing functionality and integrate Priority enum

- Added task editing modal in task component
- Integrated Priority enum for task priorities
- Updated task service to handle task updates
- Enhanced task-list component to support task editing

**Review by Fatimata-dev — APPROVED** [review]

File: src/app/components/task/task.component.ts [hunk 1]
```diff
@@ -1,24 +1,31 @@
...
   @Input() public task !: Task;
   
   @Output()
   deleteTaskToParent: EventEmitter<Task> = new EventEmitter();
+  @Output()
+  editTaskToParent: EventEmitter<Task> = new EventEmitter();
+
 
   toggleTaskComplete(){
     //this.taskService.toggleTaskComplete(this.task);
```
Comment (Benja-san) [inline src/app/components/task/task.component.ts:29]: dead code

File: src/app/service/tasks/task.service.ts [hunk 2]
```diff
@@ -21,6 +21,10 @@ export class TaskService {
      return this.httpClient.delete<Task>(`${this.apiUrl}/expeditions/${expId}/tasks/${taskId}`)
   }
 
+  updateTask(task: Task, expId: number): Observable<Task> {
+    return this.httpClient.put<Task>(`${this.apiUrl}/expeditions/${expId}/tasks/${task.id}`, task);
+  }
+
   // toggleTaskComplete(task : Task) : void {
```
Comment (Benja-san) [inline src/app/service/tasks/task.service.ts:28]: dead code

**Review by Benja-san — CHANGES_REQUESTED** [review]

File: src/app/service/tasks/task.service.ts (hunk 2 above)
Comment (Benja-san) [reply src/app/service/tasks/task.service.ts:28]: still dead code ?

**Review by Benja-san — APPROVED** [review]

**Pull request merged**